service/src/main/java/com/commercetools/pspadapter/payone/mapping/MappingUtil.java

```diff
@@ -66,48 +78,39 @@ public static void mapBillingAddressToRequest(
         request.setZip(billingAddress.getPostalCode());
         request.setCity(billingAddress.getCity());
         request.setEmail(billingAddress.getEmail());
-        request.setTelephonenumber(Optional
-                .ofNullable(billingAddress.getPhone())
-                .orElse(billingAddress.getMobile()));
+        request.setTelephonenumber(
+                ofNullable(billingAddress.getPhone())
+                        .orElse(billingAddress.getMobile()));
 
         if (countriesWithStateAllowed.contains(billingAddress.getCountry())) {
             request.setState(billingAddress.getState());
         }
     }
 
-    public static void mapCustomerToRequest(final AuthorizationRequest request, final Reference<Customer> customerReference) {
+    public static void mapCustomerToRequest(@Nonnull final AuthorizationRequest request,
+                                            @Nullable final Reference<Customer> customerReference) {
 
         Preconditions.checkArgument(customerReference != null && customerReference.getObj() != null, "Missing customer object");
         final Customer customer = customerReference.getObj();
 
         request.setVatid(customer.getVatId());
 
         //birthday
-        Optional.ofNullable(customer.getDateOfBirth()).ifPresent(birthday -> {
-            request.setBirthday(birthday.format(DateTimeFormatter.ofPattern("yyyyMMdd")));
-        });
-
-        //TODO: Gender can also be a CustomField enum @Cart with values Female/Male
-        //gender
-        Optional.ofNullable(customer.getCustom()).ifPresent(customs -> {
-            Optional.ofNullable(customs.getFieldAsString(CustomFieldKeys.GENDER_FIELD))
-                    .ifPresent(gender -> {
-                        request.setGender(gender.substring(0, 0));
-                    });
-        });
+        ofNullable(customer.getDateOfBirth())
+                .ifPresent(birthday -> request.setBirthday(dateToBirthdayString(birthday)));
 
         //customerNumber
-        Optional.ofNullable(customer.getCustomerNumber())
-            .ifPresent(customerNumber -> {
-                if (customerNumber.length() > 20) {
-                    LOG.warn("customer.customerNumber exceeds the maximum length of 20! Using substring of customer.id as fallback.");
-                    String id = customer.getId();
-                    id = id.replace("-", "").substring(0, 20);
-                    request.setCustomerid(id);
-                } else {
-                    request.setCustomerid(customerNumber);
-                }
-            });
+        ofNullable(customer.getCustomerNumber())
+                .ifPresent(customerNumber -> {
+                    if (customerNumber.length() > 20) {
+                        LOG.warn("customer.customerNumber exceeds the maximum length of 20! Using substring of customer.id as fallback.");
```

Comment:
it is: https://github.com/commercetools/commercetools-payone-integration/blob/master/docs/Field-Mapping.md#commercetools-payment-resource-mapping